es/index.js

```diff
@@ -18,6 +18,20 @@ export function isNetworkError(error) {
 const SAFE_HTTP_METHODS = ['get', 'head', 'options'];
 const IDEMPOTENT_HTTP_METHODS = SAFE_HTTP_METHODS.concat(['put', 'delete']);
 
+/**
+ * Makes a stateful adapter that keeps track of the current retry state.
+ * @param {AxiosRequestConfig} origConfig
+ * @param {Object} currentState
+ * @return {Function}
+ */
+function retryAdapterFactory(origConfig, currentState) {
+  const { adapter: origAdapter } = origConfig;
+  function adapter(config) {
```

Comment:
In particular this feels potentially problematic to me, if folks depend on `config.adapter` being the same instance of the adapter function that they pass in, this might break. Alternatively, mutating the adapter that consumers pass in also feels p gross. Happy to discuss this.

Amidst the discussion, I'm happy to write tests for these to make this PR pass CI.